misc/format-checker.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+rev=HEAD^
+clang_format=clang-format
+
+while getopts ":f:r:" opt; do
+	case $opt in
+	f)
+		clang_format=$OPTARG
+		;;
+	r)
+		rev=$OPTARG
+		;;
+	\?)
+		echo "Usage: $0 [options]"
+		echo ""
+		echo "Options:"
+		echo "  -f  Specify a command to use as a formatter [default=clang-format]"
+		echo "  -r  Specify rev to use as a base commit (any file changed after rev will be checked) [default=HEAD^]"
+		exit 1
+	esac
+done
+
+cd `git rev-parse --show-toplevel`
+files=`git diff --cached --name-only ${rev} | egrep -v '(^deps/|/_|^handler/mimemap/defaults\.c\.h)' | grep -e '.*\.[ch]\(\.in\)\?$'`
```

Comment:
Is my understanding correct that this is the only line that have code specific to the h2o/h2o repository?

Assuming that is the case, how about
* moving the logic that specifies the files outside of the shell script (e.g., to misc/docker-ci/check.mk), and
* embedding format-checker.sh in the docker image
so that this docker image can be used for applying clang-format to other repositories too (e.g., h2o/picotls, h2o/quicly)?

Reply:
👍 That sounds great.

I was actually wondering if I should really create a separate Docker image (instead I could simply write commands to `.travis.yml`) but now with your idea having a Docker image makes much more sense.

Let me try that idea and get back to you.